message.go

```diff
@@ -442,7 +447,7 @@ func alloc(s *Segment, sz Size) (*Segment, address, error) {
 		var err error
 		s, err = s.msg.allocSegment(sz)
 		if err != nil {
-			return nil, 0, err
+			return nil, 0, fmt.Errorf("allocSegment failed: %v", err)
```

Comment:
We previously purged `fmt` form the code-base in order to avoid using reflection, which helps keep binary sizes small.  I know it's a bit awkward, but can we replace this with `errors.New("allocSegment failed: " + err.Error())`?

Reply:
Ah, I wasn't aware of it. It would be an improvement (IMO) then to create specific errors for the possible error causes and then manually wrap them (e.g. similar to how https://github.com/decred/dcrd/blob/master/internal/blockchain/error.go has `ErrorKind` and `ContextError`) which would remove the reliance on `fmt` and allow a finer grained check for errors.

But that's a project for the future... :P